src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationService.java

```diff
@@ -391,4 +406,25 @@ public List<Conversation> findAllConversationsForNotifications(User user, boolea
         }
         return conversationRepository.findAllWhereUserIsParticipant(user.getId());
     }
+
+    /**
+     * Filter all channels where the attached lecture/exercise has been released
+     *
+     * @param channels A stream of channels
+     * @return A stream of channels for lectures/exercises that have been released
+     */
+    public Stream<Channel> filterVisibleChannelsForStudents(Stream<Channel> channels) {
+        ZonedDateTime now = ZonedDateTime.now();
+        return channels.filter(channel -> {
+            if (channel.getExercise() != null) {
+                return channel.getExercise().getReleaseDate() == null || channel.getExercise().getReleaseDate().isBefore(now);
+            }
+            else if (channel.getLecture() != null) {
+                return channel.getLecture().getStartDate() == null || channel.getLecture().getStartDate().isBefore(now);
```

Comment:
I think lectures are still visible if the start date is in the future, so this should not be filtered here.

Also I think the exam visibility date needs to be checked for exam-channels as well.

Reply:
You are right. I removed the filter for lectures and added the start date filter for exams instead.